### PR TITLE
feat: failure injector

### PR DESCRIPTION
**Pull request by 0xgleb**
## What

Adds a `FailureInjector` mechanism gated behind the `test-support` feature flag, along with a new `docs/feature-flags.md` reference document covering the full feature flag inventory and guidance on when to use `cfg(test)` vs `feature = "test-support"`.

## Why

End-to-end tests need a way to force conductor jobs to fail terminally without modifying production code paths. A dedicated injector type, invisible in production builds, provides this capability cleanly.

## How

`FailureInjector` wraps an `Arc<AtomicBool>`. Calling `arm()` sets the flag; the next call to `is_armed()` atomically swaps it back to `false` and returns `true`, ensuring single-use semantics and automatic disarming. The type is `Clone` so it can be shared across worker registrations.

In `test-support` builds, the apalis `work` handler gains an additional `Data<FailureInjector>` parameter. When armed, the handler logs a terminal failure and returns early without invoking `Job::perform`. In production builds, the original single-parameter `work` signature is used unchanged via `#[cfg(not(feature = "test-support"))]`.

`FailureInjector` is threaded through `ConductorCtx`, `Ctx`, and both worker registrations (order-fill and hedge) using `#[cfg(feature = "test-support")]` guards so no production code is affected.

The feature flags documentation covers the full flag inventory, the distinction between `cfg(test)` and `feature = "test-support"`, common dead-code warning pitfalls, and the zero-size-in-prod / functional-in-test struct pattern.

## Testing

Unit tests added directly in `conductor/job.rs` (gated on `test-support`) verify:

- `FailureInjector` is not armed by default.
- Arming and then checking auto-disarms on the first read.
- Arming via one clone is visible from another clone, and is consumed exactly once.

## Anything else

Several files contain unresolved merge conflict markers (`<<<<<<< HEAD` / `>>>>>>>`) between the `rest_api`/`redemption_wallet` fields and the new `failure_injector` field. These conflicts need to be resolved before this branch can be merged.

## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive feature-flags guide covering test-support, mock, wallet flags, unit vs e2e guidance, gating pitfalls (dead_code), a decision table, and a pattern for zero-size-in-production / functional-in-tests.

* **New Features**
  * Test-only, feature-gated failure-injection for job workers to simulate terminal failures; exposed to tests when enabled.

* **UI Improvements**
  * Improved log/message rendering, dialog backdrop behavior, click-handler consistency, list keys, and minor event parsing.

* **Accessibility**
  * Added aria-label for a status toggle.

### CODE REVIEW

**Comment by 0xgleb** [comment]
* **#626** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/626?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#627](https://github.com/ST0x-Technology/st0x.liquidity/pull/627) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/627?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#632](https://github.com/ST0x-Technology/st0x.liquidity/pull/632) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/632?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#625** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/625?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/625?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=625).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/conductor/builder.rs [hunk 1]
```diff
@@ -55,6 +57,8 @@ pub(crate) struct ConductorCtx<Prov, Exec> {
     pub(crate) poll_notify: Arc<tokio::sync::Notify>,
     pub(crate) wallet_polling: Option<WalletPollingCtx>,
     pub(crate) tokenizer: Option<Arc<dyn Tokenizer>>,
+    #[cfg(feature = "test-support")]
+    pub(crate) failure_injector: FailureInjector,
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply src/conductor/builder.rs:61]: now it's scoped per job kind

File: dashboard/src/lib/components/log-panel.svelte [hunk 2]
```diff
@@ -387,21 +387,17 @@
...
+      .map(([key, value]) => `${key}=${value}`)
       .join(' ')
-  }
 
   const getTarget = (entry: LogEntry): string => {
     const spans = entry.spans?.map(span => span.name).join(' > ')
     if (spans) return `${entry.target}::${spans}`
     if (entry.span) return `${entry.target}::${entry.span.name}`
-    return entry.target ?? ''
+    return entry.target
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply dashboard/src/lib/components/log-panel.svelte]: was flagged by linter

File: dashboard/src/lib/components/log-panel.svelte [hunk 3]
```diff
@@ -387,21 +387,17 @@
...
   }
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply dashboard/src/lib/components/log-panel.svelte]: was flagged by linter

File: docs/feature-flags.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+## Flag inventory
+
+| Flag                 | Purpose                             | Enabled by    |
+| -------------------- | ----------------------------------- | ------------- |
+| `test-support`       | Test infra visible to e2e tests     | dev-deps only |
+| `mock`               | Mock executor / EVM implementations | dev-deps only |
+| `wallet-turnkey`     | Turnkey wallet support              | `all-wallets` |
+| `wallet-private-key` | Local signer wallet support         | `all-wallets` |
+| `all-wallets`        | All wallet backends                 | `default`     |
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply docs/feature-flags.md]: not needed

**Review by findolor — APPROVED** [review]

**Review by JuaniRios — APPROVED** [review]
